src/GenLevelStudiesHists.cc

```diff
@@ -73,11 +83,36 @@ void GenLevelStudiesHists::fill(const RecoEvent & event){
 
   hist<TH1F>("NumberOfParticles")->Fill(event.genparticles_all->size(),weight);
 
-  int NChargedParticles = 0;                                   // Initialise counter
+  int NChargedParticles = 0;
+  TLorentzVector FourLeptons;                                   // Initialise counter
   for(size_t i=0; i<event.genparticles_all->size(); i++){
     GenParticle m = event.genparticles_all->at(i);
     // hist<TH1F>("PdgId")->Fill(m.pdgid(), weight);
     hist<TH1F>("PdgId2")->Fill(m.pdgid(), weight);
+
+    if (m.pdgid() == 25 && m.get_statusflag(GenParticle::isLastCopy))  {
+      if (!m.get_statusflag(GenParticle::isHardProcess) && !m.get_statusflag(GenParticle::fromHardProcessBeforeFSR)) continue;
+      hist<TH1F>("verifH")->Fill(m.m(), weight);
+      for(size_t j=0; j<event.genparticles_all->size(); j++) {
+        GenParticle z = event.genparticles_all->at(j);
+        if (z.mother_identifier() == m.identifier()) {
+          hist<TH1F>("verifZ")->Fill(z.m(), weight);
+          for(size_t k=0; k<event.genparticles_all->size(); k++) {
+            GenParticle l = event.genparticles_all->at(k);
+            if (l.mother_identifier() == z.identifier()) {
+              FourLeptons += l.p4();
+            }
+          }
+        }
+      }
+    }
+    TLorentzVector zero;
+    if (FourLeptons != zero) {
+      hist<TH1F>("4Lmass")->Fill(FourLeptons.M(), weight);
+      hist<TH1F>("4Lpt")->Fill(FourLeptons.Pt(), weight);
+    }
+
+
```

Comment:
This is fine. Another way is to fill a vector with the identifier of the 4 leptons then create the 4L 4vector and fill the hist if there are exactly 4 indices.